src/main/java/shef/servlets/NewCommentServlet.java

```diff
@@ -31,14 +35,24 @@
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Entity userCommentEntity = new Entity("UserComment");
+
+    // Check if the user is logged in.
+    // If so, they must have an account. So, get the user's key,
+    // as a string. If not, leave the key-string property blank.
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String id = userService.getCurrentUser().getUserId();
+      Key userKey = KeyFactory.createKey("User", id);
+      String keyString = KeyFactory.keyToString(userKey);
+      userCommentEntity.setProperty("key-string", keyString);
```

Comment:
I was a little confused as to what this key-string property holds, because at first I thought it represented the comment's key instead of the user's. I think changing it to "user-key" or maybe even just "user" would help;

Reply:
Fixed